include/utils/str.h

```diff
@@ -3,44 +3,196 @@
 
 #include <stddef.h>
 #include <stdarg.h>
+#include <stdbool.h>
 
-typedef struct str_t {
+typedef struct {
     size_t len;
     char* data;
 } str;
 
-void str_create(str *string);
-void str_create_from_cstr(str *string, const char *cstr);
-void str_create_from_data(str *string, const char *data, size_t len);
+/**
+ * @brief Create an empty string object.
+ * @details Creates a new, empty string object. After creating the object, it must be freed
+ *          using the str_free. 
+ * @param dst Target string object
+ */
+void str_create(str *dst);
+
+/**
+ * @brief Create a new string from existing string.
+ * @details Source string contents will be copied.
+ * @param dst Target string object
+ * @param src Source string to copy
+ */
+void str_from(str *dst, const str *src);
+
+/**
+ * @brief Create a string object from C string (null terminated buffer).
+ * @details Source string contents will be copied.
+ * @param dst Target string object
+ * @param src Source C string to copy
+ */
+void str_from_c(str *dst, const char *src);
```

Comment:
The old name ```str_from_cstr``` was better, since this ```str_from_c``` makes you think string from one character

Reply:
I like the str_from_c better, because:
- Faster to type
- Means we are "importing" from pure C side, so we can use this suffix in other places too if we need to
- I kept misreading cstr as just str for some reason

Reply:
Yeah, i guess it's ok